app/models/game.rb

```diff
@@ -6,10 +6,15 @@ class Game < ActiveRecord::Base
   belongs_to :black_player, class_name: "User"
   
   def find_piece_icon(x,y)
-    piece = self.pieces.where( :x_position => x, :y_position => y).first
+    #piece = self.pieces.where( :x_position => x, :y_position => y).first
+    piece = piece_on?(x,y)
     return (!piece.nil?) ? piece.icon.html_safe : ""
   end
   
+  def piece_on?(x,y)
```

Comment:
? - notation is best per convention used to return true/false values.. the method returns a piece though..